pkg/querier/worker/worker.go

```diff
@@ -71,6 +83,15 @@ type processor interface {
 	notifyShutdown(ctx context.Context, conn *grpc.ClientConn, address string)
 }
 
+// serviceDiscoveryNotifications is the interface implemented by the component receiving service discovery notifications.
+type serviceDiscoveryNotifications interface {
+	AddressAdded(address string)
+	AddressRemoved(address string)
+}
```

Comment:
We're already importing `schedulerdiscovery`, so why not using `schedulerdiscovery.Notifications` interface here?

Reply:
See: https://github.com/grafana/mimir/pull/2957#issuecomment-1250875565

> What's about worker.serviceDiscoveryNotifications? The querier worker need to discover either query-frontends or query-schedulers. I could have just used schedulerdiscovery.Notifications interface, but looks queried using it for query-frontends discovery (how is query-scheduler related at all if you're not running it?) so I preferred to duplicate it given wasn't too complex.

Reply:
> I think we can just reuse it.

Ok. Let's see the next PR which will change this interface, and then we can take a better decision on how to proceed on this interface.